File: app/src/main/java/com/nptel/courses/online/utility/GenericResponse.java
package com.nptel.courses.online.utility;

import com.google.gson.annotations.SerializedName;

public class GenericResponse<T> {

    private boolean success;
    private String status;
    @SerializedName("status_code")
    private int statusCode;
    @SerializedName("status_message")
    private String statusMessage;
    @SerializedName("error_message")
    private String errorMessage;
    @SerializedName("error_name")
    private String errorName;
    @SerializedName("error_data")
    private String errorData;
    private T data;

    public boolean isSuccess() {
        return success;
    }

    public void setSuccess(boolean success) {
        this.success = success;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public int getStatusCode() {
        return statusCode;
    }

    public void setStatusCode(int statusCode) {
        this.statusCode = statusCode;
    }

    public String getStatusMessage() {
        return statusMessage;
    }

    public void setStatusMessage(String statusMessage) {
        this.statusMessage = statusMessage;
    }

    public String getErrorMessage() {
        return errorMessage;
    }

    public void setErrorMessage(String errorMessage) {
        this.errorMessage = errorMessage;
    }

    public String getErrorName() {
        return errorName;
    }

    public void setErrorName(String errorName) {
        this.errorName = errorName;
    }

    public String getErrorData() {
        return errorData;
    }

    public void setErrorData(String errorData) {
        this.errorData = errorData;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }
}

File: app/src/main/java/com/nptel/courses/online/utility/GenericResponse.kt
package com.nptel.courses.online.utility

import com.google.gson.annotations.SerializedName

class GenericResponse<T> {
    var isSuccess = false
    var status: String? = null

    @SerializedName("status_code")
    var statusCode = 0

    @SerializedName("status_message")
    var statusMessage: String? = null

    @SerializedName("error_message")
    var errorMessage: String? = null

    @SerializedName("error_name")
    var errorName: String? = null

    @SerializedName("error_data")
    var errorData: String? = null
    var data: T? = null
        private set

    fun setData(data: T) {
        this.data = data
    }
}